ARExperiment/ARExperiment/RecognizeObjectsViewController.swift

```diff
@@ -112,54 +114,48 @@ class RecognizeObjectsViewController: UIViewController {
     
     func visionRequestDidComplete(request: VNRequest, error: Error?) {
         self.semaphore.signal()
-        var objectRect: CGRect = .zero
 
         if usingTinyModel {
-            let (predictionRect, predictionName) = processPredictionFromTinyYolo(results: request.results)
-            objectName = predictionName
-            objectRect = predictionRect ?? .zero
+            prediction = processPredictionFromTinyYolo(results: request.results)
         } else {
-            let (predictionRect, predictionName) = processPredictionFromVision(results: request.results)
-            objectName = predictionName
-            objectRect = predictionRect ?? .zero
+            prediction = processPredictionFromVision(results: request.results)
         }
-        addBoxOnMainThread(objectRect)
-
+        addBoxOnMainThread(prediction)
     }
 
-    private func processPredictionFromTinyYolo(results: [Any]?) -> (CGRect?, String) {
+    private func processPredictionFromTinyYolo(results: [Any]?) -> ObjectPrediction? {
```

Comment:
this is awesome! now it is a bit more obvious that the entire prediction is optional - before it looked like only its position good be optional (the rect was optional but the string not). So now there's no accidental confusion 💯